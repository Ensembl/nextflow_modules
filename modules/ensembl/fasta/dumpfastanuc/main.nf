// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

process FASTA_DUMPFASTANUC {
    tag "${db.species}"
    label 'process_low'
    conda "${moduleDir}/environment.yml"
    container 'ensemblorg/ensembl-legacy-scripts:e112_APIv0.4'

    input:
        val(db)

    output:
        tuple val(db), val("fasta_dna"), path("*.fasta"), emit: nucleotide_fasta
        path "versions.yml", emit: versions

    when:
        task.ext.when == null || task.ext.when

    script:
        def prefix = task.ext.prefix ?: "${db.server.database}"
        def version = "0.4"
        output = "${db.species}_fasta_dna.fasta"
        password_arg = db.server.password ? "--pass ${db.server.password}" : ""

        """
        dump_fasta_dna.pl \
            --host ${db.server.host} \
            --port ${db.server.port} \
            --user ${db.server.user} \
            ${password_arg} \
            --dbname ${db.server.database} \
            > ${output}
        
        echo -e -n "${task.process}:\n\tensembl-genomio: ${version}" > versions.yml
        """

    stub:
        def args = task.ext.args ?: ''
        def prefix = task.ext.prefix ?: "${db.test_id}"
        def version = "0.4"

        output_file = "dna.fasta"
        dump_dir = "${workflow.projectDir}/tests/modules/ensembl/fasta/dump_nucleotide/"
        dump_file = "dumped_dna.fasta"
        """
        cp ${dump_dir}/${dump_file} ${output_file}
        echo -e -n "${task.process}:\n\tensembl-genomio: ${version}" > versions.yml
        """
}
