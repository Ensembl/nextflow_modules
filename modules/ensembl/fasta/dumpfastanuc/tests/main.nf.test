// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// nf-core modules test fasta/dumpfastanuc
nextflow_process {

    name "Test Process FASTA_DUMPFASTANUC"
    script "../main.nf"
    process "FASTA_DUMPFASTANUC"

    tag "modules"
    tag "modules_local"
    tag "fasta"
    tag "fasta/dumpfastanuc"

    test("Fasta Dump Nucleotide stub") {
        when {

            options "-stub"

            process {
                """
                input[0] = Channel.of(
                    '>SEQ1',
                    'CGTA'
                    )
                    .collectFile(name: 'dna.fasta', newLine: true)
                    .map { file -> [ [ id:'test dump dna' ], file ] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}