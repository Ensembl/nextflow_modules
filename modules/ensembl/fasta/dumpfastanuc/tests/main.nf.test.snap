{
    "Fasta Dump Nucleotide [stub]": {
        "content": [
            {
                "0": [
                    [
                        "dummy_core_db_core_110_1",
                        "fasta_dna",
                        "dna.fasta:md5,22af65c79ff5f3d36e7ff1c6d6dcdd6b"
                    ]
                ],
                "nucleotide_fasta": [
                    [
                        "dummy_core_db_core_110_1",
                        "fasta_dna",
                        "dna.fasta:md5,22af65c79ff5f3d36e7ff1c6d6dcdd6b"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-11-29T14:35:44.892449722"
    },
    "nucleotide_fasta": {
        "content": [
            [
                [
                    "dummy_core_db_core_110_1",
                    "fasta_dna",
                    "dna.fasta"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-11-29T14:35:44.835653261"
    }
}